### PR TITLE
feat: standalone targeting, fractional shorthand

### PR DESCRIPTION
**Pull request by toddbaert**
This PR does 2 things:

- Allows for fractional targeting shorthand where no weight (assumes weight = 1)
- Allows the targeting.json schema to be used alone, not just as part of the flag definition schema

Non functionally, it restructures our testing so that tests pertaining to `targeting.json` exist in the `test/targeting/` dir, while tests pertaining to the `flags.json` exist in the `test/flags` dir.

Fixes: https://github.com/open-feature/flagd-schemas/issues/165

### CODE REVIEW

File: json/targeting.json [hunk 1]
```diff
@@ -462,7 +467,7 @@
       "$comment": "if we remove the \"sum to 100\" restriction, update the descriptions below!",
       "description": "Distribution for all possible variants, with their associated weighting out of 100.",
       "type": "array",
-      "minItems": 2,
+      "minItems": 1,
```
Comment (toddbaert) [inline json/targeting.json:470]: This fixes the fractional issue (no weight = 1).

File: json/test/positive/custom-ops.json [hunk 2]
```diff
@@ -25,6 +25,21 @@
...
+      "state": "ENABLED",
+      "variants": {
+        "heads": "heads",
+        "tails": "tails",
+        "draw": "draw"
+      },
+      "defaultVariant": "draw",
+      "targeting": {
+        "fractional": [
+          [ "heads" ],
```
Comment (toddbaert) [inline json/test/positive/custom-ops.json:38]: This was invalid before.

File: json/test/targeting-only/positive/basic-json-ops.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
+{
+  "$schema": "../../../targeting.json",
+  "targeting": {
+    "*" : [
```
Comment (toddbaert) [inline json/test/targeting-only/positive/basic-json-ops.json:4]: We don't need a ton of tests here, just 1 positive and 1 negative (above) to makes sure `targeting` can be used in isolation. The rest is covered with the definitions tests.

File: json/targeting.yaml [hunk 4]
```diff
@@ -4,6 +4,9 @@ title: flagd Targeting
 description: Defines targeting logic for flagd; a extension of JSONLogic, including
   purpose-built feature-flagging operations.
 type: object
+properties: 
+  targeting:
+    $ref: "#/definitions/targeting"
```
Comment (toddbaert) [inline json/targeting.yaml:9]: This allows for the use of targeting by itself.

File: json/test/targeting/negative/fractional-invalid-bucketing.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (toddbaert) [inline json/test/targeting/negative/fractional-invalid-bucketing.json:1]: All these tests used to be embedded in `flag` json objects, which was redundant. This PR converts all tests which just test targeting validation to purely targeting tests, since we can now validate targeting alone.

File: json/targeting.json (hunk 1 above)
Comment (toddbaert) [inline json/targeting.json:470]: This allows for the fractional shorthand support (no wieght == weight 1)

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by aepfli — CHANGES_REQUESTED** [review]
so far the code and test changes do look good, but i do think we should change the schema changes, to not need the property

File: json/targeting.json [hunk 6]
```diff
@@ -4,6 +4,11 @@
   "title": "flagd Targeting",
   "description": "Defines targeting logic for flagd; a extension of JSONLogic, including purpose-built feature-flagging operations.",
   "type": "object",
+  "properties": {
+    "targeting": {
+      "$ref": "#/definitions/targeting"
+    }
+  },
```
Comment (aepfli) [inline json/targeting.json:11]:
```suggestion
 "anyOf": [
   {
     "$ref": "#/definitions/targeting"
   }
 ],
```

Comment (toddbaert) [reply json/targeting.json]: Oh you're totally right. I've corrected this and updated all the tests!

File: json/targeting.json [hunk 7]
```diff
@@ -4,6 +4,11 @@
   "title": "flagd Targeting",
   "description": "Defines targeting logic for flagd; a extension of JSONLogic, including purpose-built feature-flagging operations.",
   "type": "object",
+  "properties": {
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply json/targeting.json]: Absolutely correct, thanks a lot.

I've made this change.

**Review by aepfli — APPROVED** [review]
